debug-ui/app/components/Orderbook.tsx

```diff
@@ -25,25 +25,61 @@ const Orderbook = ({
   const { wallet } = useWallet();
 
   useEffect(() => {
-    const updateOrderbook = async (): Promise<void> => {
-      try {
-        const market: Market = await fetchMarket(
-          conn,
-          new PublicKey(marketAddress),
-        );
-        setCurrentSlot(market['slot']);
+    const accountChangeListenerId: number = conn.onAccountChange(
+      new PublicKey(marketAddress),
+      (accountInfo: AccountInfo<Buffer>) => {
+        setMarketData(accountInfo.data);
+      },
+    );
+
+    return () => {
+      conn.removeAccountChangeListener(accountChangeListenerId);
+    };
+  }, [marketAddress]);
+
+  useEffect(() => {
+    try {
+      if (marketData) {
+        const market: Market = Market.loadFromBuffer({
+          address: new PublicKey(marketAddress),
+          buffer: marketData,
+          slot: currentSlot,
+        });
         const asks: RestingOrder[] = market.asks();
         const bids: RestingOrder[] = market.bids();
         setBids(bids.reverse());
         setAsks(asks);
-      } catch (e) {
-        console.error('updateOrderbook:', e);
-        toast.error(`updateOrderbook: ${ensureError(e).message}`);
+      }
+    } catch (e) {
+      console.error('updateOrderbook:', e);
+      toast.error(`updateOrderbook: ${ensureError(e).message}`);
+    }
+  }, [conn, currentSlot, marketData, marketAddress]);
+
+  useEffect(() => {
+    // Initial load of the market.
+    const initialLoad = async (): Promise<void> => {
+      try {
+        const marketInfo: AccountInfo<Buffer> = (await conn.getAccountInfo(
+          new PublicKey(marketAddress),
+        ))!;
+        setMarketData(marketInfo.data);
+      } catch (err) {
+        console.error('initialLoad:', err);
       }
     };
+    initialLoad();
 
-    updateOrderbook();
-    const id = setInterval(updateOrderbook, 2_000);
+    const updateSlot = async (): Promise<void> => {
+      try {
+        setCurrentSlot(await conn.getSlot());
+      } catch (err) {
+        console.error('getSlot:', err);
+      }
+    };
+    updateSlot();
+    // 200 ms is half a slot, so we can expect this to update on each slot.
+    const id = setInterval(updateSlot, 200);
```

Comment:
could use a slot subscription, 5rps for get slot from a public UI can quickly become a few 100rps